code/modules/reagents/chemistry/machinery/reagentgrinder.dm

```diff
@@ -64,334 +48,475 @@
+		context[SCREENTIP_CONTEXT_LMB] = "Transfer contents"
+	else
+		context[SCREENTIP_CONTEXT_LMB] = "Insert item"
+	return CONTEXTUAL_SCREENTIP_SET
 
 /obj/machinery/reagentgrinder/examine(mob/user)
 	. = ..()
 	if(!in_range(user, src) && !issilicon(user) && !isobserver(user))
 		. += span_warning("You're too far away to examine [src]'s contents and display!")
 		return
 
-	if(operating)
-		. += span_warning("\The [src] is operating.")
-		return
+	var/total_weight = 0
+	var/list/obj/item/to_process = list()
+	for(var/obj/item/target in src)
+		if((target in component_parts) || target == beaker)
+			continue
+		to_process["[target.name]"] += 1
+		total_weight += target.w_class
+	if(to_process.len)
+		. += span_notice("Currently holding")
+		for(var/target_name as anything in to_process)
+			. += span_notice("[to_process[target_name]] [target_name]")
+		. += span_notice("Filled to <b>[round((total_weight / maximum_weight) * 100)]%</b> capacity")
+
+	if(!QDELETED(beaker))
+		. += span_notice("A beaker of <b>[beaker.reagents.maximum_volume]u</b> capacity is present")
+		. += span_notice("[EXAMINE_HINT("Right click")] with empty hand to remove beaker")
+	else
+		. += span_warning("Its missing an beaker")
 
-	if(beaker || length(holdingitems))
-		. += span_notice("\The [src] contains:")
-		if(beaker)
-			. += span_notice("- \A [beaker].")
-		for(var/i in holdingitems)
-			var/obj/item/O = i
-			. += span_notice("- \A [O.name].")
-
-	if(!(machine_stat & (NOPOWER|BROKEN)))
-		. += "[span_notice("The status display reads:")]\n"+\
-		span_notice("- Grinding reagents at <b>[speed*100]%</b>.")
-		if(beaker)
-			for(var/datum/reagent/R in beaker.reagents.reagent_list)
-				. += span_notice("- [R.volume] units of [R.name].")
+	. += span_notice("You can drag an storage item to dump its contents in the grinder")
+	if(anchored)
+		. += span_notice("It can be [EXAMINE_HINT("wrenched")] loose")
+	else
+		. += span_warning("Needs to be [EXAMINE_HINT("wrenched")] in place to work")
+	. += span_notice("Its maintainence panel can be [EXAMINE_HINT("screwed")] [panel_open ? "closed" : "open"]")
+	if(panel_open)
+		. += span_notice("It can be [EXAMINE_HINT("pried")] apart")
 
-/obj/machinery/reagentgrinder/attack_hand_secondary(mob/user, list/modifiers)
+/obj/machinery/reagentgrinder/update_overlays()
 	. = ..()
-	if(. == SECONDARY_ATTACK_CANCEL_ATTACK_CHAIN)
-		return
-	if(operating || !user.can_perform_action(src, ALLOW_SILICON_REACH | FORBID_TELEKINESIS_REACH))
-		return
-	eject(user)
-	return SECONDARY_ATTACK_CANCEL_ATTACK_CHAIN
 
-/obj/machinery/reagentgrinder/attack_robot_secondary(mob/user, list/modifiers)
-	return attack_hand_secondary(user, modifiers)
+	if(!QDELETED(beaker))
+		. += "[base_icon_state]-beaker"
 
-/obj/machinery/reagentgrinder/attack_ai_secondary(mob/user, list/modifiers)
-	return attack_hand_secondary(user, modifiers)
+	if(anchored && !panel_open && is_operational && !(machine_stat & (BROKEN | NOPOWER)))
```

Comment:
you should only need to check is operational because both of those flags should make it non operational